src/c++/perf_analyzer/concurrency_worker.cc

```diff
@@ -209,6 +202,8 @@ ConcurrencyWorker::AsyncCallbackFinalize(uint32_t ctx_id)
     notified_ = true;
   }
 
+  total_ongoing_requests_--;
```

Comment:
I believe this is back to how it used to be in main. It is unclear if this change solved a race condition around total_ongoing_requests vs notified_ updates, but better to be safe than sorry. Hopefully TMA-1044 can help test and expose potential races